doc_src/cmds/path.rst

```diff
@@ -0,0 +1,413 @@
+   # So it passes.
+   /home/me
+
+   >_ path filter -fdxw /usr/bin/fish /home/me
+   # This is the same as above: "-f" is "--type=file", "-d" is "--type=dir",
+   # "-x" is short for "--perm=exec" and "-w" short for "--perm=write"!
+   /home/me
+   
+.. _cmd-path-is:
+
+"is" subcommand
+--------------------
+
+::
+
+    path is [(-z | --null-in)] [(-Z | --null-out)] [(-q | --quiet)] [(-t | --type) TYPE] [(-p | --perm) PERMISSION] [PATH...]
+
+``path is`` is short for ``path filter -q``. It returns true if any of the given files passes the filter, but does not produce any output.
+
+``--quiet`` can still be passed for compatibility but is redundant. The options are the same as for ``path filter``.
+
+Examples
+^^^^^^^^
+
+::
+
+   >_ path is /usr/bin /usr/argagagji
+   # /usr/bin exists, so this returns a status of 0 (true). It prints nothing.
+   >_ path is /usr/argagagji
+   # /usr/argagagji does not, so this returns a status of 1 (false). It also prints nothing.
+   >_ path is -fx /bin/sh
+   # /bin/sh is usually an executable file, so this returns true.
+
+"expand" subcommand
+--------------------
+
+::
+
+    path expand [(-z | --null-in)] [(-Z | --null-out)] [(-q | --quiet)] [(-t | --type) TYPE] [(-p | --perm) PERMISSION] [GLOB...]
+
+``path expand`` expands the given arguments as globs, similar to other shells or the ``glob(3)`` function. It returns all the resulting paths, after optionally running them through the same filters as ``path filter``.
+
+It understands more special characters than fish's :ref:`wildcard expansion <expand-wildcard>`:
+
+- ``*`` is any number of characters, except for ``/`` and a leading ``.`` or a ``.`` after a ``/``.
+- ``**`` is any number of characters, including ``/`` and ``.``.
+- ``?`` is a single character.
+- ``[abc]`` is a character set - any character of "a", "b", or "c".
+- ``[a-z]`` is a character range - here the characters from a to z (which characters they are is locale-dependent!).
+- ``[0-9]`` is the numbers from 0 to 9.
+- ``[ac-h]`` is a set and a range, combined.
+- ``[!f-h]`` or ``[^f-h]`` is an inverted character set or range - any character except for one from f to h.
+- ``[[:alnum:]]`` is the class of alphanumeric characters. Other classes include: "cntrl" for control characters, "digit" for numeric digits, "print" for printable characters (including space).
+
+Any matched directories will be listed with a trailing "/".
+
+It is not recommended to pass full paths to ``path expand`` because they might be expanded again. If you already have a set of paths, use ``path filter`` instead.
+
+``path expand`` supports the same filter options - ``--type`` and ``--perm`` as ``path filter``, see there for an explanation. ``--invert`` is currently not supported because it is unclear what it should mean in this context.
+
+``path expand`` is limited to 512 * 1024 results per-glob, like fish's globs in general. If a glob exceeds this, ``path expand`` will print an error and as many results as it can.
```

Comment:
I think it's better to print no result in this case, to make sure that the user notices the problem and doesn't carry on with inconsistent data.

Reply:
The idea here was that you could at least run e.g.

```fish
rm (path expand)
```

and it would work for the first X files. And then you'd rerun it for the next.

It seems a bit awkward to have 512 * 1024 - 1 results, and everything works, and then you add one more result and get *nothing* instead?

Honestly, the limit here is quite awkward whatever we do. We have to have one for the command substitution this might be in (ARG_MAX being a thing), but there's no reason

```fish
path expand | xargs rm
```

shouldn't work for an arbitrary number of files. But removing the limit would require a bit of a rearchitecting of wildcard,  so I've put it off.